### PR TITLE
Replaced istanbul with nyc

### PR DESCRIPTION
**Pull request by hknustwmf**
Linted tests and added test specific lint config.

Bug:  T216191

### CODE REVIEW

File: test/functional/invalid_requests.js [hunk 1]
```diff
@@ -1,60 +1,57 @@
...
 
 var router = module.parent.router;
 var utils = require('../utils/test_utils.js');
 var deepEqual = utils.deepEqual;
 var P = require('bluebird');
 
-describe('Invalid request handling', function() {
-    before(function () { return router.setup(); });
+describe('Invalid request handling', function () {
+    before(function () { return router.setup(); }); // eslint-disable-line max-statements-per-line
```
Comment (Pchelolo) [inline test/functional/invalid_requests.js]: if we convert it to arrow func we wouldn't need to disable eslint

File: test/functional/multi_ranged.js [hunk 2]
```diff
@@ -1,16 +1,13 @@
...
 var router = module.parent.router;
 var utils = require('../utils/test_utils.js');
 var deepEqual = utils.deepEqual;
 
-describe("Multiranged tables", function() {
+describe('Multiranged tables', function () {
     this.timeout(15000);
 
-    before(function () { return router.setup(); });
+    before(function () { return router.setup(); }); // eslint-disable-line
```
Comment (Pchelolo) [inline test/functional/multi_ranged.js]: idem

File: test/functional/invalid_requests.js (hunk 1 above)
Comment (hknustwmf) [reply test/functional/invalid_requests.js]: Changed to arrow function

File: test/functional/multi_ranged.js (hunk 2 above)
Comment (hknustwmf) [reply test/functional/multi_ranged.js]: changed to arrow func

**Pull request merged**